external_samples/port.py

```diff
@@ -19,20 +19,25 @@
 from enum import Enum
 from typing import Self
 
+_base_compound = 256
```

Comment:
I think this should be all caps to show it is supposed to be a constant and shouldn't be changed.  We should probably also import Final from typing and use it here.